common/src/main/java/com/microsoft/identity/common/internal/controllers/TokenCommand.java

```diff
@@ -74,7 +71,7 @@ public AcquireTokenResult execute() throws Exception {
                         (AcquireTokenSilentOperationParameters) getParameters()
                 );
 
-                if (result.getSucceeded()) {
+                if (result != null && result.getSucceeded()) {
```

Comment:
How can it be that we returned `null` but _didn't_ throw an Exception? That would seem to indicate some bad behavior at a lower level, no?

Reply:
Agreed. Investigating further , my primary suspicion could of null here : https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/dev/msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java#L187

Per logs Broker returned the right exception and adapter did transform to the right exception. However we have some logic here https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/dev/msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java#L206 where we could miss to throw an Exception after catching and return null result. I will sync with @rpdome to understand this better